plugins/node/opentelemetry-instrumentation-knex/src/instrumentation.ts

```diff
@@ -0,0 +1,234 @@
+    );
+
+    module.files.push(
+      this.getClientNodeModuleFileInstrumentation('src'),
+      this.getClientNodeModuleFileInstrumentation('lib'),
+      this.getRunnerNodeModuleFileInstrumentation('src'),
+      this.getRunnerNodeModuleFileInstrumentation('lib'),
+      this.getRunnerNodeModuleFileInstrumentation('lib/execution')
+    );
+
+    return module;
+  }
+
+  private getRunnerNodeModuleFileInstrumentation(basePath: string) {
+    return new InstrumentationNodeModuleFile<typeof knex>(
+      `knex/${basePath}/runner.js`,
+      constants.SUPPORTED_VERSIONS,
+      (Runner: any, moduleVersion) => {
+        api.diag.debug(
+          `Applying runner.js patch for ${constants.MODULE_NAME}@${moduleVersion}`
+        );
+        this.ensureWrapped(
+          moduleVersion,
+          Runner.prototype,
+          'query',
+          this.wrapQuery.bind(this)
+        );
+        return Runner;
+      },
+      (Runner: any, moduleVersion) => {
+        api.diag.debug(
+          `Removing patch for ${constants.MODULE_NAME}@${moduleVersion}`
+        );
+        this._unwrap(Runner.prototype, 'query');
+        return Runner;
+      }
+    );
+  }
+
+  private getClientNodeModuleFileInstrumentation(basePath: string) {
+    return new InstrumentationNodeModuleFile<typeof knex>(
+      `knex/${basePath}/client.js`,
+      constants.SUPPORTED_VERSIONS,
+      (Client: any, moduleVersion) => {
+        api.diag.debug(
+          `Applying client.js patch for ${constants.MODULE_NAME}@${moduleVersion}`
+        );
+        this.ensureWrapped(
+          moduleVersion,
+          Client.prototype,
+          'queryBuilder',
+          this.storeContext.bind(this)
+        );
+        this.ensureWrapped(
+          moduleVersion,
+          Client.prototype,
+          'schemaBuilder',
+          this.storeContext.bind(this)
+        );
+        this.ensureWrapped(
+          moduleVersion,
+          Client.prototype,
+          'raw',
+          this.storeContext.bind(this)
+        );
+        return Client;
+      },
+      (Client: any, moduleVersion) => {
+        api.diag.debug(
+          `Removing client.js patch for ${constants.MODULE_NAME}@${moduleVersion}`
+        );
+        this._unwrap(Client.prototype, 'queryBuilder');
+        this._unwrap(Client.prototype, 'schemaBuilder');
+        this._unwrap(Client.prototype, 'raw');
+        return Client;
+      }
+    );
+  }
+
+  private wrapQuery(original: Function) {
+    const instrumentation = this;
+    return function wrapped_logging_method(this: any, query: any) {
+      const config = this.client.config;
+
+      const maxLen = (instrumentation._config as types.Config).maxQueryLength!;
+      const attributes: any = {
+        'knex.version': instrumentation._moduleVersion,
+        [SemanticAttributes.DB_SYSTEM]: utils.mapSystem(config.client),
+      };
+      if (maxLen !== 0) {
+        attributes[SemanticAttributes.DB_STATEMENT] = utils.limitLength(
+          query?.sql,
```

Comment:
according to the `?`, the str param to `limitLength` can be undefined, but then in the function you are calling `str.substr` which will throw.
I think it check for undefined here or in the function

Reply:
:+1: That's called false sense of safety from Typescript.